### PR TITLE
refactor: show the code intel status indicator via Sourcegraph extensions

### PR DESCRIPTION
**Pull request by chrismwendt**
See https://github.com/sourcegraph/sourcegraph-langserver-http/pull/7

# Before

# After

Fixes https://github.com/sourcegraph/sourcegraph/issues/236
Fixes https://github.com/sourcegraph/enterprise/issues/13632

> This PR does not need to update the CHANGELOG because this doesn't change the UX much

### CODE REVIEW

File: cmd/frontend/registry/backcompat.go [hunk 1]
```diff
@@ -113,8 +113,27 @@ func init() {
...
 				// https://github.com/sourcegraph/sourcegraph-langserver-http.
-				Url:              "https://storage.googleapis.com/sourcegraph-cx-dev/sourcegraph-langserver-http.3.js",
+				Url:              "https://storage.googleapis.com/sourcegraph-cx-dev/sourcegraph-langserver-http.4.js",
 				ActivationEvents: []string{"onLanguage:" + ls.Language},
+				Contributes: &schema.Contributions{
+					Actions: []*schema.Action{
+						&schema.Action{
+							Id: "langserver.status",
+							ActionItem: &schema.ActionItem{
+								Description: "Code intelligence active for ${resource.language}",
```
Comment (felixfbecker) [inline cmd/frontend/registry/backcompat.go]: Before the status indicator showed what features can be expected to work. This is a regression

Comment (sqs) [reply cmd/frontend/registry/backcompat.go]: This is an acceptable feature regression. We will remedy it in the future when each language's extension is a separate extension and can offer more language-specific advice about what works and what doesn't (similar to how in VS Code, the Go/Python extensions tell you to install certain tools that are necessary and can report friendlier error messages about why things don't work).

File: schema/extension.schema.json [hunk 2]
```diff
@@ -59,15 +64,56 @@
...
+              }
+            }
+          }
+        },
+        "menus": {
+          "type": "object",
+          "properties": {
+            "editor/title": { "type": "array", "items": { "$ref": "#/definitions/MenuItem" } },
+            "commandPalette": { "type": "array", "items": { "$ref": "#/definitions/MenuItem" } },
+            "help": { "type": "array", "items": { "$ref": "#/definitions/MenuItem" } }
```
[the remark this answers is not in the document]
Comment (chrismwendt) [reply schema/extension.schema.json]: Added descriptions

File: schema/extension.schema.json [hunk 3]
```diff
@@ -59,15 +64,56 @@
...
+            "properties": {
+              "id": { "type": "string" },
+              "command": { "type": "string" },
+              "commandArguments": {
+                "type": "array",
+                "items": { "type": "string" }
+              },
+              "title": { "type": "string" },
+              "category": { "type": "string" },
+              "iconURL": { "type": "string" },
```
[the remark this answers is not in the document]
Comment (chrismwendt) [reply schema/extension.schema.json]: I can't change `iconURL` in this PR

File: cmd/frontend/registry/backcompat.go [hunk 4]
```diff
@@ -113,8 +113,27 @@ func init() {
...
+								IconURL:     "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB2aWV3Qm94PSIwIDAgMjQgMjQiPgogICAgICAgICAgICAgICAgICAgPHN2ZyBjbGFzcz0ibWRpLWljb24gaWNvbi1pbmxpbmUiIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgZmlsbD0iIzM3YjI0ZCIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgICAgICAgICAgICAgICAgICAgPHBhdGggZD0iTTE2LDdWM0gxNFY3SDEwVjNIOFY3SDhDNyw3IDYsOCA2LDlWMTQuNUw5LjUsMThWMjFIMTQuNVYxOEwxOCwxNC41VjlDMTgsOCAxNyw3IDE2LDdaIj48L3BhdGg+CiAgICAgICAgICAgICAgICAgICA8L3N2Zz4KICAgICAgICAgICAgICAgICA8L3N2Zz4KICAgICAgICAgICAgICAgIA==",
```
[the remark this answers is not in the document]
Comment (chrismwendt) [reply cmd/frontend/registry/backcompat.go]: Added a comment

File: schema/extension.schema.json [hunk 5]
```diff
@@ -59,12 +64,102 @@
       "description":
         "Features contributed by this extension. Extensions may also register certain types of contributions dynamically.",
       "type": "object",
-      "additionalProperties": false,
```
Comment (felixfbecker) [inline schema/extension.schema.json]: Why does this now allow additional properties?

File: schema/extension.schema.json [hunk 6]
```diff
@@ -59,12 +64,102 @@
...
+                "type": "string"
+              },
+              "command": {
+                "description": "The command to execute when this action is taken.",
+                "type": "string"
+              },
+              "commandArguments": {
+                "description": "The arguments to the command.",
+                "type": "array",
+                "items": { "type": "string" }
```
Comment (felixfbecker) [inline schema/extension.schema.json]: Is there a reason why commands can only receive string arguments?

File: schema/extension.schema.json [hunk 7]
```diff
@@ -59,12 +64,102 @@
...
+              },
+              "title": {
+                "description": "The text that is shown in the UI.",
+                "type": "string"
+              },
+              "category": {
+                "description": "The category of this action.",
```
Comment (felixfbecker) [inline schema/extension.schema.json]: This description is not very helpful. How is the category used? What would be an example?

File: schema/extension.schema.json [hunk 8]
```diff
@@ -59,12 +64,102 @@
...
+              "actionItem": {
+                "description": "The action item.",
+                "type": "object",
+                "properties": {
+                  "label": {
+                    "description": "The text that is shown on the action item the UI.",
+                    "type": "string"
+                  },
+                  "description": {
+                    "description": "The tooltip text for an action item that is shown in the UI.",
```
Comment (felixfbecker) [inline schema/extension.schema.json]: Mention that this is a template string?

**Comment by chrismwendt** [comment]
Addressed the rest of the feedback in #365

**Pull request merged**